File: debop4k-benchmark/src/main/java/debop4k/benchmark/models/Cell.java
package debop4k.benchmark.models;//package debop4k.benchmark.models;
//
//
//import lombok.Data;
//
//import java.io.Serializable;
//
//
//public class Cell implements Serializable {
//
//  public static Cell of() {
//    return new Cell();
//  }
//
//  public static Cell of(int r, int c) {
//    return new Cell(r, c);
//  }
//
//  public Cell() {this(0, 0);}
//
//  public Cell(int r, int c) {
//    this.row = r;
//    this.col = c;
//  }
//
//  private int row = 0;
//  private int col = 0;
//
//  private String text = null;
//  private byte[] bytes = null;
//
//  public int getRow() {
//    return row;
//  }
//
//  public int getCol() {
//    return col;
//  }
//
//
//  public String getText() {
//    return text;
//  }
//
//  public void setText(String text) {
//    this.text = text;
//  }
//
//  public byte[] getBytes() {
//    return bytes;
//  }
//
//  public void setBytes(byte[] bytes) {
//    this.bytes = bytes;
//  }
//}

File: debop4k-benchmark/src/test/kotlin/debop4k/benchmark/models/Cell.kt
/*
 *  Copyright (c) 2016. KESTI co, ltd
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package debop4k.benchmark.models

import debop4k.core.utils.hashOf
import java.io.Serializable

data class Cell @JvmOverloads constructor(val row: Int = 0,
                                          val col: Int = 0,
                                          var text: String? = null,
                                          var bytes: ByteArray? = null) : Serializable {

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    return other != null && other is Cell && hashCode() == other.hashCode()
  }

  override fun hashCode(): Int {
    return hashOf(row, col)
  }

  companion object {
    @JvmStatic
    @JvmOverloads
    fun of(r: Int = 0, c: Int = 0): Cell = Cell(r, c)
  }
}